packages/react-reconciler/src/ReactFiberHotReloading.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {ReactElement} from 'shared/ReactElementType';
+import type {Fiber} from './ReactFiber';
+import type {FiberRoot} from './ReactFiberRoot';
+
+import {
+  flushSync,
+  scheduleWork,
+  flushPassiveEffects,
+} from './ReactFiberScheduler';
+import {Sync} from './ReactFiberExpirationTime';
+import {
+  FunctionComponent,
+  ForwardRef,
+  MemoComponent,
+  SimpleMemoComponent,
+} from 'shared/ReactWorkTags';
+import {REACT_FORWARD_REF_TYPE, REACT_MEMO_TYPE} from 'shared/ReactSymbols';
+
+type Family = {|
+  currentType: any,
+  currentSignature: null | string,
+|};
+
+type HotUpdate = {|
+  root: FiberRoot,
+  updatedFamilies: Set<Family>,
+  staleFamilies: Set<Family>,
+|};
+
+type HotReloadingInterface = {|
+  scheduleHotUpdate: HotUpdate => void,
+|};
+
+// Avoid any overhead even in DEV when hot reloading is off:
+export let resolveTypeForHotReloading = (type: any): any => type;
+export let isCompatibleFamilyForHotReloading = (
+  fiber: Fiber,
+  element: ReactElement,
+) => false;
+export let isInvalidatedForHotReloading = (fiber: Fiber | null): boolean =>
```

Comment:
Closure is likely smart enough to inline this, recognize that it always returns false, and strip it out of the conditional check, but relying on seems a bit flaky.

Separately, `export let` could cause Rollup to wrap this module in a closure.

A more predictable pattern would be:

```js
export function isInvalidatedForHotReloading(type) {
  // supportHotReloading is injected by DevTools hook
  if (__DEV__ && supportsHotReloading) {
    // Implementation body
  }
}
```

This will compile to a no-op in prod, which is less likely to confuse Closure.

Reply:
Addressed.